src/pybind/rados/rados.pyx

```diff
@@ -889,6 +889,13 @@ Rados object in state %s." % self.state)
         if ret >= 0:
             return int(ret)
         elif ret == -errno.ENOENT:
+            rev = self.wait_for_latest_osdmap() 
```

Comment:
seems like we should handle this within librados itself, rather than the python bindings - in RadosClient::lookup_pool()